packages/kit/src/exports/vite/index.js

```diff
@@ -351,6 +373,13 @@ function kit({ svelte_config }) {
 						vite_config_env.command === 'serve' ? env.private : undefined
 					);
 				case '\0$env/dynamic/public':
+					// populate `$env/dynamic/public` from `window` in the case where
+					// modules are externally hosted
+					if (kit.paths.assets && is_build && !options?.ssr) {
+						// TODO use a hash of the version name to avoid conflicts
+						return `export const env = window.__sveltekit_${version_hash}.env;`;
```

Comment:
In #8901 we also write to the global window. We should put these things below a common top level key. I went with `$__sveltekit__`. We can put the hash at the top level (like you do) or one level below.

Btw the TODO (and one below where you set the env) is probably outdated.

Reply:
Good catch. I think top-level is probably easiest since we then don't need to check to see if `$__sveltekit__` already exists before creating it/adding the key

Reply:
Is there a potential use case in the future (programmatic navigation from outside for example) that warrants a stable key? What conflicts do you had in mind when creating this? Multiple embedded SvelteKit apps?

Reply:
Yeah, I was thinking about embeds mainly

Reply:
removed the TODO comments